tests/phpunit/CRM/Civicase/Hook/SelectWhereClause/LimitCaseQueryToAccessibleCaseCategoriesTest.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+use CRM_Civicase_Test_Fabricator_Contact as ContactFabricator;
+use CRM_Civicase_Test_Fabricator_CaseType as CaseTypeFabricator;
+use CRM_Civicase_Test_Fabricator_Case as CaseFabricator;
+
+/**
+ * Tests for limiting case query to accessible case categories.
+ *
+ * @group headless
+ */
+class CRM_Civicase_Hook_SelectWhereClause_LimitCaseQueryToAccessibleCaseCategoriesTest extends BaseHeadlessTest {
+
+  /**
+   * Fabricated Client.
+   *
+   * @var array
+   */
+  private static $client;
+
+  use CRM_Civicase_Helpers_SessionTrait;
+
+  /**
+   * Setup case types and cases before running of tests.
+   */
+  public static function setupBeforeClass() {
+    civicrm_api3('OptionValue', 'create', [
+      'option_group_id' => 'case_type_categories',
+      'name' => 'award',
+      'value' => 2,
+    ]);
+    $caseType = CaseTypeFabricator::fabricate([
+      'name' => 'Case_type_first',
+      'case_type_category' => 1,
+
+    ]);
+    $caseType2 = CaseTypeFabricator::fabricate([
+      'name' => 'Case_type_second',
+      'case_type_category' => 2,
+    ]);
+    static::$client = ContactFabricator::fabricate();
+
+    CaseFabricator::fabricate(
+      [
+        'case_type_id' => $caseType['id'],
+        'contact_id' => static::$client['id'],
+        'creator_id' => static::$client['id'],
+      ]
+    );
+    CaseFabricator::fabricate(
+      [
+        'case_type_id' => $caseType2['id'],
+        'contact_id' => static::$client['id'],
+        'creator_id' => static::$client['id'],
+      ]
+    );
+  }
+
+  /**
+   * Test that no records are returned if user has no permissions.
+   */
+  public function testNoCasesAreReturnedForNoPermissions() {
+    $contact = ContactFabricator::fabricate();
+    $this->registerCurrentLoggedInContactInSession($contact['id']);
+    $this->setPermissions(['access my cases and activities']);
```

Comment:
We don't really need to set a permission here, the test says when no permission, you can just pass an empty array.

Reply:
Actually I need this permission to call for case api that is called in the same test below